### ISSUE TITLE
Problem with enum resolution

### ISSUE THREAD

**Issue by retailcoder** [rubberduck-vba/Rubberduck#2447]
Seems enum types don't get resolved anymore.

**Class1**
````vb
Public Enum TestEnum
    Foo
    Bar
End Enum
````

**Class2**
````vb
Public Function GetValue() As TestEnum
    GetValue = Foo
End Function
````

The `TestEnum` in the function's `AsTypeClause` doesn't get resolved to the enum type, and thus the `GetValue = Foo` assignment spawns an incorrect "Object variable 'GetValue' is assigned without the 'Set' keyword" ...which should really differenciate between a function/property and a variable, but that's another issue.

**Comment by comintern** [comment]
I think this might be an issue with how scoping is handled on these.  I noticed this when I was trying to match how the `ReferencedDeclarationsCollector` assigns qualified names to how user declarations are qualified.  It stuck me as odd at the time, but Enum members are scoped to the *module*, and not to the enumeration itself.  This in itself a resolver bug, because RD picks the first identifier that it encounters instead of making sure it's enclosed in the appropriate Enum:

![untitled](https://cloud.githubusercontent.com/assets/8944005/21473743/c1cc4fc6-cada-11e6-8b0e-dec7cf45cca5.png)

**Comment by ThunderFrame** [comment]
That doesn't compile: Ambiguous name detected.

**Comment by comintern** [comment]
Right, but RD is currently Enum treating references like that *should* compile. This was giving me hash collisions when I was trying to add Declarations to a Dictionary, which should never happen.  My guess is that the namespace resolution for Enums is borked in general.

**Comment by ThunderFrame** [comment]
hmm, do we need an inspection for this?

**Module1**
```
Enum Foo
  X = 1
  Y = 2
End Enum

Function FooBar() As Bar
  FooBar = X
End Function
```

**Module 2**
```
Option Explicit

Public Enum Bar
  X = 3
  Y = 4
End Enum
```

**Comment by retailcoder** [comment]
Fixed in 027103a2
